web-common/src/features/chat/core/chat.ts

```diff
@@ -0,0 +1,596 @@
+        if (event.query.queryKey.toString() === queryKey.toString()) {
+          const updatedData = event.query.state.data as
+            | V1GetConversationResponse
+            | undefined;
+          const updatedMessages = updatedData?.conversation?.messages || [];
+          set(updatedMessages);
+        }
+      });
+
+      return unsubscribe;
+    },
+    [] as V1Message[],
+  );
+
+  // =============================================================================
+  // SHARED GLOBAL MUTATIONS
+  // =============================================================================
+
+  // Mutation for creating a new conversation with the first message
+  const completeNewConversation = createRuntimeServiceComplete(
+    {
+      mutation: {
+        onMutate: async (variables) => {
+          const { instanceId, data } = variables;
+
+          // Create optimistic conversation ID and cache key
+          const optimisticConversationId = getOptimisticConversationId();
+          const optimisticCacheKey = getConversationCacheKey(
+            instanceId,
+            optimisticConversationId,
+          );
+
+          // Create user message from the mutation data
+          const userMessage: V1Message = {
+            id: getOptimisticMessageId(),
+            role: data.messages?.[0]?.role || "user",
+            content: data.messages?.[0]?.content || [],
+            createdOn: new Date().toISOString(),
+            updatedOn: new Date().toISOString(),
+          };
+
+          // Create optimistic conversation in cache
+          queryClient.setQueryData(
+            optimisticCacheKey,
+            (): V1GetConversationResponse => ({
+              conversation: {
+                id: optimisticConversationId,
+                createdOn: new Date().toISOString(),
+                updatedOn: new Date().toISOString(),
+                messages: [userMessage],
+              },
+            }),
+          );
+
+          // Update current conversation ID to optimistic ID
+          currentConversationId.set(optimisticConversationId);
+
+          return { optimisticConversationId };
+        },
+
+        onSuccess: async (response, variables, context) => {
+          const { instanceId } = variables;
+
+          if (response.conversationId && context?.optimisticConversationId) {
+            // Fetch the specific conversation to get complete data including title
+            const conversationResponse = await queryClient.fetchQuery(
+              getRuntimeServiceGetConversationQueryOptions(
+                instanceId,
+                response.conversationId,
+              ),
+            );
+
+            // Add the new conversation to the `listConversations` TanStack Query Cache
+            const listQueryKey =
+              getRuntimeServiceListConversationsQueryKey(instanceId);
+            queryClient.setQueryData(listQueryKey, (oldList: any) => {
```

Comment:
Could you add type to `oldList` and `conv` in this block? Hard to follow otherwise why we merge conversations field and the object above separately.

Reply:
Good call– updated